ibis/backends/sql/rewrites.py

```diff
@@ -188,7 +188,7 @@ def extract_ctes(node):
     g = Graph.from_bfs(node, filter=(ops.Relation, ops.Subquery, ops.JoinLink))
     for node, dependents in g.invert().items():
         if isinstance(node, ops.View) or (
-            len(dependents) > 1 and isinstance(node, cte_types)
+            isinstance(node, cte_types) and len(dependents) > 1
```

Comment:
Why reverse the check order here?

Reply:
Can you leave this as is? There's no obvious reason to churn this code right now.

Reply:
Sure, I was just discovering this issue while trying to extract mssql subqueries with sortby expressions and I had another instancecheck here which looked easier to understand. Going to restore.

Reply:
```suggestion
            len(dependents) > 1 and isinstance(node, cte_types)
```